File: shared/src/test/java/ru/privatenull/pnauth/security/TotpKeyStoreTest.java
package ru.privatenull.pnauth.security;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.io.TempDir;

import java.nio.file.Files;
import java.nio.file.Path;

import static org.junit.jupiter.api.Assertions.assertArrayEquals;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

class TotpKeyStoreTest {
    @Test
    void createsAndReusesOneKey(@TempDir Path directory) throws Exception {
        Path keyFile = directory.resolve("totp.key");
        byte[] created = TotpKeyStore.loadOrCreate(keyFile);

        assertEquals(32, created.length);
        assertArrayEquals(created, TotpKeyStore.loadOrCreate(keyFile));
    }

    @Test
    void rejectsMalformedExistingKey(@TempDir Path directory) throws Exception {
        Path keyFile = directory.resolve("totp.key");
        Files.write(keyFile, new byte[10]);

        assertThrows(java.io.IOException.class, () -> TotpKeyStore.loadOrCreate(keyFile));
    }
}

File: shared/src/test/kotlin/ru/privatenull/pnauth/security/TotpKeyStoreTest.kt
package ru.privatenull.pnauth.security

import org.junit.jupiter.api.Assertions.assertArrayEquals
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertThrows
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.io.TempDir
import java.io.IOException
import java.nio.file.Files
import java.nio.file.Path

class TotpKeyStoreTest {
    @Test
    fun createsAndReusesOneKey(@TempDir directory: Path) {
        val keyFile = directory.resolve("totp.key")
        val created = TotpKeyStore.loadOrCreate(keyFile)

        assertEquals(32, created.size)
        assertArrayEquals(created, TotpKeyStore.loadOrCreate(keyFile))
    }

    @Test
    fun rejectsMalformedExistingKey(@TempDir directory: Path) {
        val keyFile = directory.resolve("totp.key")
        Files.write(keyFile, ByteArray(10))

        assertThrows(IOException::class.java) { TotpKeyStore.loadOrCreate(keyFile) }
    }
}
